llvm/lib/MC/MCGOFFStreamer.cpp

```diff
@@ -32,3 +40,71 @@ MCStreamer *llvm::createGOFFStreamer(MCContext &Context,
     S->getAssembler().setRelaxAll(true);
   return S;
 }
+
+void MCGOFFStreamer::initSections(bool NoExecStack,
+                                  const MCSubtargetInfo &STI) {
+  MCContext &Ctx = getContext();
+  if (NoExecStack)
+    switchSection(Ctx.getAsmInfo()->getNonexecutableStackSection(Ctx));
+  else
+    switchSection(Ctx.getObjectFileInfo()->getTextSection());
+}
+
+void MCGOFFStreamer::switchSection(MCSection *S, const MCExpr *Subsection) {
+  auto Section = cast<MCSectionGOFF>(S);
+  MCSection *Parent = Section->getParent();
+
+  if (Parent) {
+    const MCExpr *Subsection = Section->getSubsectionId();
+    assert(Subsection && "No subsection associated with child section");
+    this->MCObjectStreamer::switchSection(Parent, Subsection);
+    return;
```

Comment:
So this collapses sections back into their parent sections?   I guess I'm confused why we need to create those separate sections then in the first place.  Couldn't we just use the parent section and appropriate subsection to begin with?

Reply:
Also, talking about subsections - I wonder if it would make sense to model the GOFF SD vs. ED relationship as section vs. subsection?  That might allow to pull policy logic out of the GOFF writer and into the places that define/use sections ...

Reply:
One reason we didn't want to do that is because an SD Record, on its own, cannot own any text. (ie. the "owner" field of a TXT record cannot be the EsdId of a SD record), so in this case a section wouldn't actually be allowed to own text except indirectly through its subsections.